### PR TITLE
Adding backend related methods to `QiskitRuntimeLocalService`

### LINKED ISSUE #1606: Add backend related methods to QiskitRuntimeLocalService
**What is the expected feature or enhancement?**

`QiskitRuntimeLocalService` was created to support local testing mode and is meant to be similar to `QiskitRuntimeService`. The initial release of `QiskitRuntimeLocalService` only has the `run()` method, and we immediately got feedback asking for the other methods 🙂. 

This issue is about adding backend related methods to `QiskitRuntimeLocalService`. 

**Acceptance criteria**

Add the following methods to `QiskitRuntimeLocalService`
- backends
- backend
- least_busy

In local testing mode, the only backend accessible are the fake backends, hence these methods should only return fake backends (e.g. `backend("fake_sherbrooke")` would work but not `backend("ibm_sherbrooke")`)

### PR DESCRIPTION
**Pull request by SamFerracin**
### Summary
Add the following methods to QiskitRuntimeLocalService

backends
backend
least_busy

### Details and comments
Fixes #1606

### CODE REVIEW

File: qiskit_ibm_runtime/fake_provider/local_service.py [hunk 1]
```diff
@@ -51,6 +54,91 @@ def __init__(
...
+
+        Args:
+            min_num_qubits: Minimum number of qubits the fake backend has to have.
+            filters: More complex filters, such as lambda functions, that can be defined as for the
+                :meth:`backends` method.
+
+        Returns:
+            A fake backend.
+        """
+        return choice(self.backends(min_num_qubits=min_num_qubits, filters=filters))
```
Comment (SamFerracin) [inline qiskit_ibm_runtime/fake_provider/local_service.py:140]: @jyu00 this is my best guess of what this function should do, but let me know if you have any thoughts. For example, should it pick a random backend the first time, and then just keep returning that same backend?

**Review by kt474 — COMMENTED** [review]
Can you also add a release note - instructions are here https://github.com/Qiskit/qiskit-ibm-runtime/blob/main/CONTRIBUTING.md#release-notes

You can see examples [here](https://github.com/Qiskit/qiskit-ibm-runtime/blob/main/release-notes/0.24.0.rst), these later get rendered in the qiskit docs [here](https://docs.quantum.ibm.com/api/qiskit-ibm-runtime/release-notes)

File: release-notes/0.24.1.rst [hunk 2]
```diff
@@ -1,5 +1,9 @@
 0.24.1 (2024-06-18)
 ===================
+New Features
+------------
+
+- Added ``backend``, ``backends``, and ``least_busy`` methods to ``QiskitRuntimeLocalService``. (`1764 <https://github.com/Qiskit/qiskit-ibm-runtime/pull/1764>`__)
```
Comment (kt474) [inline release-notes/0.24.1.rst:6]: Sorry I gave a bad example here - this `0.24.1.rst` file is what the release notes look like after they are generated during a release. For PRs, the release notes should be in the format `<PR number>.<type>.rst`. So for example, for this PR, it'd be: `1764.feat.rst`

Better examples: https://github.com/jyu00/qiskit-ibm-runtime/tree/a2f6df81c74f9ed9cbaeba7ae75fbd18668d6da7/release-notes/unreleased

Comment (SamFerracin) [reply release-notes/0.24.1.rst]: Nice automation flow!

**Review by kt474 — APPROVED** [review]

**Comment by jyu00** [comment]
It's too late now since this was already released, but just for future reference, `QiskitRuntimeLocalService` is actually a undocumented class (i.e. it doesn't show up in API reference). It's hidden because it was not meant to be instantiated directly. So one wouldn't know about its existence and thus its new backend methods unless they look at the code, at least until #1608 is released that is.

**Pull request merged**